src/main/resources/category/java/common_std.xml

```diff
@@ -87,4 +87,66 @@ public class StringStuff {
         </example>
     </rule>
 
+    <rule name="LetFieldsMeetSerializable"
+          language="java"
+          message="Fields in a Serializable class should be serializable or transient."
+          class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
+          externalInfoUrl="${doc_root}/JavaCodeQuality.md#isr02">
+        <description>Problem: Field in a Serializable class is not serializable nor transient. When (de)serialization happens, a RuntimeException will be thrown and (de)serialization fails.
+            Solution: make the field either transient, make its class implement Serializable or interface extend Serializable.
+            Note: Classes extending Throwable do, by inheritance, implement Serializable, yet are excluded in this rule, since they are typically never actually serialized.
+            An exception to this exception is when extending RemoteException, then fields should be transient or serializable.
+        </description>
+        <priority>2</priority>
+        <properties>
+            <property name="tags" value="jpinpoint-rule,pitfall,replaces-sonar-rule" type="String" description="classification"/>
+            <property name="xpath">
+                <value><![CDATA[
+//ClassDeclaration[ImplementsList/ClassType[@SimpleName='Serializable']
+or pmd-java:typeIs('java.io.Serializable')
+(: Throwable is the exception, except RemoteException :)
+and not(pmd-java:typeIs('java.lang.Throwable') and not(pmd-java:typeIs('java.rmi.RemoteException')))]
+(: non-transient, non-static, non-primitive fields :)
+//FieldDeclaration[not(pmd-java:modifiers() = 'transient') and not(pmd-java:modifiers() = 'static')]
```

Comment:
ik denk dat je dit ook korter kunt schrijven als:
not(pmd-java:modifiers() = ('transient', 'static') )
Zie bijv. AvoidMutableLists uit common.xml:
@MethodName = ('set', 'clear', 'replaceAll', 'retainAll')